sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -269,13 +267,14 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
     gnrc_rpl_parent_t *old_best = dodag->parents;
     gnrc_rpl_parent_t *new_best = old_best;
     uint16_t old_rank = dodag->my_rank;
-    gnrc_rpl_parent_t *elt = NULL, *tmp = NULL;
+    gnrc_rpl_parent_t *elt, *tmp;
+    kernel_pid_t if_id;
 
     if (dodag->parents == NULL) {
         return NULL;
     }
 
-    LL_FOREACH_SAFE(dodag->parents, elt, tmp) {
+    LL_FOREACH(dodag->parents, elt) {
```

Comment:
the safe version is only needed when removing entries from the list that is iterated on.